provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
Yea, I'll add respective comments. I still find it utterly useless and a tool that forces me to do useless things doesn't seem like a particularly good tool. After all the tool should help me, not the other way around.